### PR TITLE
Added support for docx in Google Drive

### PR DESCRIPTION
**Pull request by sidravi1**
I needed this for my org and may be useful to others

### CODE REVIEW

File: backend/danswer/connectors/google_drive/connector.py [hunk 1]
```diff
@@ -174,7 +192,9 @@ def _process_folder_paths(
                     service=service, parent_id=parent_id, folder_name=folder_name
                 )
                 if parent_id is None:
-                    raise ValueError(f"Folder path '{path}' not found in Google Drive")
+                    raise ValueError(
+                        f"Folder '{folder_name}' in path '{path}' not found in Google Drive"
```
Comment (sidravi1) [inline backend/danswer/connectors/google_drive/connector.py:197]: When debugging, I found it useful to know which folder was missing in the path

File: backend/danswer/connectors/google_drive/connector.py [hunk 2]
```diff
@@ -45,7 +47,10 @@ def get_folder_id(
     """
     Get the ID of a folder given its name and the ID of its parent folder.
     """
-    query = f"'{parent_id}' in parents and name='{folder_name}' and mimeType='{DRIVE_FOLDER_TYPE}'"
+    query = (
+        f"'{parent_id}' in parents and name='{folder_name}' and "
```
Comment (sidravi1) [inline backend/danswer/connectors/google_drive/connector.py:52]: Line longer than black's 88 but it doesn't fix long strings.

**Review by yuhongsun96 — APPROVED** [review]
Looks good and works, great stuff! Thanks!

File: backend/danswer/connectors/google_drive/connector.py (hunk 2 above)
Comment (yuhongsun96) [reply backend/danswer/connectors/google_drive/connector.py:52]: Will soon add github actions to auto format and detect type check issues to make things easier for contributors

**Comment by yuhongsun96** [comment]
Ah, looks to have some merge conflicts, if you could update it to latest main real quick, then I can put it in. Sorry for the extra hassle.

**Comment by sidravi1** [comment]
> Ah, looks to have some merge conflicts, if you could update it to latest main real quick, then I can put it in. Sorry for the extra hassle.

Thanks for reviewing this. Fixed the merge conflict!

**Comment by yuhongsun96** [comment]
Merged, thanks!

**Pull request merged**